proto/buf/alpha/registry/v1alpha1/display.proto

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020-2021 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "buf/alpha/registry/v1alpha1/role.proto";
+
+// DisplayService is the display service that
+// provides information regarding to UI element display.
+service DisplayService {
+  // DisplayRepositoryRolesToManage returns which roles should be displayed
+  // to the user when they are managing contributors on the repository.
+  rpc DisplayRepositoryRolesToManage(DisplayRepositoryRolesToManageRequest) returns (DisplayRepositoryRolesToManageResponse);
+  // DisplayRepositoryRolesToManageByUser returns which roles should be displayed
+  // to the user when they are managing a specific contributor on the repository.
+  rpc DisplayRepositoryRolesToManageByUser(DisplayRepositoryRolesToManageByUserRequest) returns (DisplayRepositoryRolesToManageByUserResponse);
```

Comment:
How about 
```suggestion
  // ListManagableRepositoryRoles returns which roles should be displayed
  // to the user when they are managing a specific contributor on the repository.
  rpc ListManagableRepositoryRoles(ListManagableRepositoryRolesRequest) returns (ListManagableRepositoryRolesResponse);
```

Maybe `Manageable` is enough to indicate that this is per-user, as opposed to `Assignable`, which is for adding a new role?

Reply:
I think we will still want a Manageable endpoints that is not per-user (I assume this is one of the use case for `UserCanManageRepositoryContributors` in the contributors list to determine show setting icon or not, again  @timostamm @seankimdev correct me if I am wrong)

Reply:
Ah right, so we have 3 use cases for these two RPCs:

1. What roles to display when user presses "Add User To Organization". Is this covered by `ListManageableRepositoryRoles`? I could see it being user-dependent, so maybe it should be covered by `ListManageableUserRepositoryRoles` instead?
2. On which members to display the cog wheel icon when viewing the list of members. This is covered by `ListManageableRepositoryRoles` I assume, since it's simple to do a check for the each listed users role against such a list.
3. What roles to display when _changing_ an existing users role. This is presumably covered by `ListManageableUserRepositoryRoles`, since it's user specific.

Am I missing any use cases? Maybe we should have 3 RPCs that map cleanly to each use case instead? Or is there some other way we can unify these? CC @timostamm.

Reply:
I will let Timo answer if any use case missing / incorrect as he knows better about that, but considering these 3 use cases, I think 1 and 3 share exactly the same request/response/behaviour, and in fact representing the same thing? I don't think we will need 3 RPCs if that is the case

Reply:
I concur with @cyinma - `ListManageableRepositoryRoles` for case 2, and `ListManageableUserRepositoryRoles` for cases 1 and 3.

Reply:
Sounds good to me 👍🏻